spec/support/saml_auth_helper.rb

```diff
@@ -45,6 +45,42 @@ def saml_cert
     @saml_cert ||= File.read("#{Rails.root}/certs/saml_cert.crt")
   end
 
+  # generates a SAML response and returns a parsed Nokogiri XML document
+  def generate_saml_response(options = {})
+    # set default options
+    options.reverse_merge! ial_token: false
+    # user needs to be signed in in order to generate an assertion
+    sign_in(user)
+    send_get_request(options)
+  end
```

Comment:
This method already exists down below towards the end of the file. This one up here should be removed, and the one down below should be updated to remove the begin/rescue block.